tests/vendor_tests/redhat/foreman_tests.py

```diff
@@ -0,0 +1,61 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+
+from sos_tests import StageOneReportTest, redhat_only
+
+
+class ForemanBasicTest(StageOneReportTest):
+    """Ensure that a basic execution runs as expected with all TFM related
+    plugins
+
+    :avocado: tags=foreman
+    """
+
+    sos_cmd = '-v'
+
+    def test_tfm_plugins_ran(self):
+        self.assertPluginIncluded([
+            'foreman',
+            'candlepin',
+            'puppet',
+            'ruby'
+        ])
+
+    def test_foreman_keys_skipped(self):
+        self.assertFileGlobNotInArchive("/etc/foreman*/*key.pem")
+
+    def test_foreman_database_sizes_collected(self):
+        self.assertFileCollected('sos_commands/foreman/foreman_db_tables_sizes')
+
+    def test_foreman_database_dumps_collected(self):
+        self.assertFileCollected('sos_commands/foreman/foreman_settings_table')
+
+    def test_foreman_csv_collected(self):
+        self.assertFileCollected('sos_commands/foreman/foreman_tasks_tasks')
+
+    def test_proxyfeatures_not_collected(self):
+        self.assertFileGlobNotInArchive("*/smart_proxies_features/*")
+
+    def test_candlepin_table_sizes_collected(self):
+        self.assertFileCollected('sos_commands/candlepin/candlepin_db_tables_sizes')
+
+    def test_foreman_password_postproc_worked(self):
+        # known value for our sealed test image
+        self.assertFileNotHasContent(
+            '/var/log/foreman-installer/katello.log',
+            'rs74B8sAiaEJtDzdVgu5SAP2TZHQSDS4'
+        )
```

Comment:
This is now failing (I *think* as my Satellite instance has `satellite.log` but not `katello.log`) due to still not obfuscated:

    2021-04-26 16:37:11 [INFO  ] [configure] +  password: "QaUerNfEdZg4q78Sqsw6VskRpLYxwp9B"

Reply:
I've pulled in the updates from that PR, however I had another thought.

I pulled this password directly from the logs after the test setup. Is there a standard way to control this (and other passwords?) in the foreman installation?

I was thinking this might be a pain when it comes to adding foreman 2.5 and later to the test matrix. If we can statically set an array of passwords to well-known values, then it makes maintenance easier and allows us to validate a lot more passwords.

Reply:
`foreman-installer` has option `--foreman-db-password`. Optionally, we can fetch it directly from the config file:

    /etc/foreman/database.yml:  password: "QaUerNfEdZg4q78Sqsw6VskRpLYxwp9B"

and then use that value in the `assertFileNotHasContent` test.